ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/Context.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.ethsigner.core;
+
+import tech.pegasys.ethsigner.core.http.HttpResponseFactory;
+import tech.pegasys.ethsigner.core.jsonrpc.JsonDecoder;
+import tech.pegasys.ethsigner.core.signing.TransactionSignerProvider;
+
+import java.nio.file.Path;
+import java.time.Duration;
+
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpClientOptions;
+import io.vertx.core.http.HttpServerOptions;
+
+public class Context {
+
+  private final long chainId;
+  private final TransactionSignerProvider transactionSignerProvider;
+  private final HttpClientOptions clientOptions;
+  private final Duration httpRequestTimeout;
+  private final HttpResponseFactory responseFactory = new HttpResponseFactory();
+  private final JsonDecoder jsonDecoder;
+  private final Path dataPath;
+  private final Vertx vertx;
+  private final HttpServerOptions serverOptions;
+
+  public Context(
```

Comment:
These values look like they are used specifically for either the VerticleManager or the HttpServerServiceFactory so don't think we need a context object at all. The verticleManager only needs the dataPath and vertx whereas HttpServerServiceFactory needs everything else + vertx.